controller/scheduler/swarm.py

```diff
@@ -0,0 +1,136 @@
+import re
+import time
+from docker import Client
+from django.conf import settings
+from .states import JobState
+
+MATCH = re.compile(
+    '(?P<app>[a-z0-9-]+)_?(?P<version>v[0-9]+)?\.?(?P<c_type>[a-z-_]+)?.(?P<c_num>[0-9]+)')
+
+
+class SwarmClient(object):
+    def __init__(self, target, auth, options, pkey):
+        self.target = settings.SWARM_HOST
+        # single global connection
+        self.registry = settings.REGISTRY_HOST+":"+settings.REGISTRY_PORT
+        self.docker_cli = Client(base_url='tcp://'+self.target+':'+"2395",
+                                 timeout=1200, version='1.17')
+
+    def create(self, name, image, command='', template=None, **kwargs):
+        """Create a container"""
+        cimage = self.registry+"/"+image
+        cname = name
```

Comment:
Why are we assigning a new variable without any modification? If it's just for visual clarity then we should just change the signature names. Doesn't make sense to waste space in the private heap, IMO